clash-hedgehog/src/Clash/Hedgehog/Sized/Index.hs

```diff
@@ -0,0 +1,20 @@
+{-|
+Copyright   : (C) 2021, QBayLogic B.V.
+License     : BSD2 (see the file LICENSE)
+Maintainer  : QBayLogic B.V. <devops@qbaylogic.com>
+
+Random generation of Index.
+-}
+
+module Clash.Hedgehog.Sized.Index
+  ( genIndex
+  ) where
+
+import GHC.TypeLits (KnownNat)
+import Hedgehog (MonadGen, Range)
+import qualified Hedgehog.Gen as Gen
+
+import Clash.Sized.Internal.Index
+
+genIndex :: (MonadGen m, KnownNat n) => Range (Index n) -> m (Index n)
+genIndex = Gen.integral
```

Comment:
For `Index` / `Signed` / `Unsigned` this isn't true because users control the `Range` used

Reply:
The Range just changes the extreme values from a grow/shrink perspective. So say you've got a range from 1 to 2 million, it would be _incredibly_ unlikely to actually generate that 2 million. However this is usually what you want while testing systems.